torch/csrc/jit/codegen/cuda/lower_alias_memory.cpp

```diff
@@ -61,127 +63,903 @@ class SymbolicSizePrinter : private kir::IrVisitor {
+//!   the buffer.
+//!   Outer interval marks the begining of the loop of first write and end of
+//!   the loop of last read, both at the same loop level as the buffer
+//!   allocation.
+class BufferUseDefInfo {
  public:
-  void modify(const std::vector<kir::Expr*>& exprs) {
-    // Find candidate TensorViews and collect analysis information
+  // Alias local memory if it exceeds this threshold
+  static constexpr long kRegisterSizeThreshold = 1;
+
+  BufferUseDefInfo(
+      const std::vector<kir::Expr*>& exprs,
+      BufferReuseDebugPrinter* debug_printer = nullptr)
+      : debug_printer_(debug_printer) {
+    if (debug_printer) {
+      debug_printer->buffer_info_ = this;
+    }
+    collectScopeInfo(exprs);
+    collectScopeUseDefInfo(exprs);
+  }
+
+  //! Returns live interval info of buffer if previously
+  //!  computed.
+  c10::optional<AllocationInfoPtr> getMaybeReuseInfoFor(
+      kir::Allocate* allocate) const {
+    auto alloc_it = map_allocate_to_info_.find(allocate);
+    if (alloc_it == map_allocate_to_info_.end()) {
+      return c10::nullopt;
+    }
+    auto alloc = alloc_it->second;
+    return alloc;
+  }
+
+  //! Realize alias of two buffers through inner alias analysis and
+  //!  keep track of the re-use.
+  void useInnerAlias(AllocationInfoPtr from, AllocationInfoPtr to) {
+    to->inner_alias_list_->push_back(from);
+    to->inner_subscribed_intevals->push_back(from->inner_live_interval.get());
+    setAlias(from, to);
+    from->is_inner_alias = true;
+  }
+
+  //! Realize alias of two buffers through outer alias analysis and
+  //!  keep track of the re-use.
+  void useOuterAlias(AllocationInfoPtr from, AllocationInfoPtr to) {
+    to->outer_subscribed_intevals->push_back(from->outer_live_interval.get());
+    setAlias(from, to);
+  }
+
+  //! To run before performing in-place sharing analysis.
+  //!   Initializes the inner live intervals
+  void prepareInnerSharingAnalysis() {
+    for (auto it : map_allocate_to_info_) {
+      auto alloc_info = it.second;
+      // At beginning only use interval for each
+      //  allocate is their corresponding live interval
+      alloc_info->inner_subscribed_intevals->push_back(
+          alloc_info->inner_live_interval.get());
+    }
+  }
+
+  //! To run before performing outer interval based sharing analysis.
+  //!   Initializes the outer live intervals and copy inner sharing
+  //!   information.
+  void prepareOuterSharingAnalysis() {
+    for (auto it : map_allocate_to_info_) {
+      auto alloc_info = it.second;
+      if (!alias_map_.count(alloc_info)) {
+        alloc_info->outer_subscribed_intevals->push_back(
+            alloc_info->outer_live_interval.get());
+        // Update only if this buffer isn't an alias
+        for (auto inner_alias : *(alloc_info->inner_alias_list_)) {
+          alloc_info->outer_subscribed_intevals->push_back(
+              inner_alias->outer_live_interval.get());
+        }
+      }
```

Comment:
Sure but would not be necessary since no `allocationInfo` entry will be created if the allocation is already an alias. So an alias would not become a candidate for `to_reuse` in the pass. 

Also this pass is assuming that there's no non-trivial aliasing introduced before this pass other than the ones like batch norm running mean.